src/System/Taffybar/Widget/Windows.hs

```diff
@@ -73,19 +83,37 @@ windowsNew :: WindowsConfig -> TaffyIO Gtk.Widget
 windowsNew config = do
   label <- lift $ Gtk.labelNew Nothing
 
+  icon <- lift $ Gtk.imageNew
+
+  context <- ask
+
   let setLabelTitle title = lift $ postGUIASync $ Gtk.labelSetMarkup label title
-      activeWindowUpdatedCallback _ = getActiveLabel config >>= setLabelTitle
+      getActiveWindowData = getActiveWindow >>= (traverse $ getWindowData Nothing [])
+      getThePixbuf size = fmap join $ (runX11Def Nothing getActiveWindowData) >>= (traverse $ getActiveWindowIconPixbuf config size)
+      refreshImage = flip runReaderT context . getThePixbuf
```

Comment:
I don't think this part needs a name, just put this inline in the call to autoSizeImage below.

Reply:
Done.